### ISSUE TITLE
Time picker - better UX + a11y

### ISSUE THREAD

**Issue by ThiefMaster** [indico/indico#6387]
Since I don't think it makes much sense to start a longer discussion in the comment thread of https://github.com/indico/indico/pull/6371, let's use this issue instead.

My initial comment:

> Another interesting part to look into may be the time picker. It's used in the regform when you have a date field and add time format in its config, but we have many other places where it's used as well - and while it works OK-ish most of the time, its usability is not really great, and probably its accessibility is pretty poor as well. So I'd love to eventually replace it with something nicer.

And @tomasr8's comment:

> I think the idea is to use the new https://github.com/indico/indico/pull/6310 to create a more accessible time picker

cc @foxbunny

**Comment by foxbunny** [comment]
Yeah, my suggestion was to provide a time picker that is based on the combobox with a list of time presets in 15-minute increments, similar to how it works in Google Calendar.

**Comment by ThiefMaster** [comment]
Will a combobox work for this purpose though? I think it may work OK if you can include all the times, but I'm sure you don't want to show 720 entries for every minute, and even w/ 5/15 minute steps you'd still have 288/96 individual items.

I typically just type the time - but something like `1330` for `13:30` would already require some fuzzy matching behavior. I'd certainly want the text field of the combobox to update to the properly formatted time when it loses focus (like on Google Calendar). GC also handles am/pm nicely for those using that particular format, e.g. I an enter `0130pm` and it will convert it to `1:30pm`, and if I type `1330` or `13:30` it will nonetheless recognize it and set it as `1:30pm` on blur.

All that said, I personally never found it annoying/unpleasant to enter a time on Google Calendar, and considering how popular that tool is, maybe the same kind of behavior is good enough for us and our users. :)

**Comment by axu-trex** [comment]
Is the same time picker also used in selecting durations of contributions? If so then the precision should be 5 minutes or less...

I like the time picker in some android apps. See example below. Maybe we could use something like that?

![image](https://github.com/indico/indico/assets/141645772/1c101d2d-a044-42a7-bb20-cce20eb814d9)
